components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -164,6 +164,39 @@ class GeckoEngineSession(
         // no-op (requires v63+)
     }
 
+    /**
+     * See [EngineSession.findAll]
+     */
+    override fun findAll(text: String) {
+        geckoSession.finder.find(text, 0).then { result: GeckoSession.FinderResult? ->
+            result?.let {
+                notifyObservers { onFindResult(it.current, it.total, true) }
+            }
+            GeckoResult<Void>()
+        }
+        notifyObservers { onFind(text) }
```

Comment:
Can there be a race condition here? onFind() seems to clear the list of results in `EngineObserver`. Can it be possible that `onFindResult` gets called before `onFind` here? Should we call `onFind` first?